File: SharsorIPCpp/tests/PySharsorIPC/test_consumer.py

from SharsorIPCpp.PySharsorIPC import *
import sys

# Get the total number of arguments
num_args = len(sys.argv)

# Check if there are enough arguments
if num_args < 1:
    print("Usage: python script.py [namespace]...")
    sys.exit(1)

# Access individual arguments
namespace = sys.argv[1]

consumer = Consumer(namespace=namespace,
            basename="ProducerConsumerTests",
            verbose=True,
            vlevel=VLevel.V2)

consumer.run()

timeout = 10000
terminated=False
while not terminated:
    
    try:

        if not consumer.wait(timeout):

            terminated = True
            
            break

        print("Dong stuff...")

        if not consumer.ack():

            terminated = True
            
            break

    except KeyboardInterrupt:

        terminated = True
        
        break

consumer.close()

File: SharsorIPCpp/tests/test_consumer.cpp
#include <iostream>
#include <SharsorIPCpp/Consumer.hpp>
#include <string>
#include <SharsorIPCpp/Client.hpp>
#include <SharsorIPCpp/Journal.hpp>

#include <csignal>
#include <cstdlib>

using namespace SharsorIPCpp;
using LogType = Journal::LogType;
using VLevel = Journal::VLevel;
using Consumer = SharsorIPCpp::Consumer;

std::string name_space = "rthytrtkjdhsxdhn";

int counter = 0;
int listener_idx = 0;
bool terminated = false;
unsigned int timeout = 10000;

// Signal handler function
void interruptHandler(int signal) {
    std::cout << "Interrupt signal received (Ctrl+C pressed)." << std::endl;
    terminated = true;
}

int main(int argc, char *argv[]) {

    if (argc != 2) {
        std::cerr << "Usage: " << argv[0] << " <namespace>" << std::endl;
        return 1;
        
    }
    // Get the command line argument for idx
    std::string name_space = argv[1];

    std::signal(SIGINT, interruptHandler);

    Consumer consumer = Consumer("ProducerConsumerTests", 
                            name_space,
                            true,
                            VLevel::V2);
    consumer.run();

    while(!terminated) {

        if (!consumer.wait(timeout)) {

            std::cout << "Wait failed" << std::endl;

            break;
        }

        std::cout << "Doing stuff..." << std::endl;

        if (!consumer.ack()) {

            std::cout << "Acknowledgement failed!!" << std::endl;

            break;
        }

    }

    consumer.close();

    return 0;
}

